data/finset.lean

```diff
@@ -992,6 +989,57 @@ sort_eq _ _
 
 @[simp] theorem sort_to_finset [decidable_eq α] (s : finset α) : (sort r s).to_finset = s :=
 list.to_finset_eq (sort_nodup r s) ▸ eq_of_veq (sort_eq r s)
+
 end sort
 
+variable [decidable_eq α]
```

Comment:
As a practical measure, I think it would be a good idea to wrap everything that requires `variable [decidable_eq α]` in a `section disjoint`/`end disjoint` to avoid adding the unnecessary variable to anything that gets appended later.

Reply:
done